### PR TITLE
WEB-3284 - remove invalid integration categorization

### PR DESCRIPTION
**Pull request by davidejones**
### What does this PR do?

This PR:
- Uses the classifiers in the assetlib as a reference to only allow valid categorization of integrations
- Write to the `layouts/shortcodes/integration_categories.md` file with the classifiers so that they can be used by the `https://docs.datadoghq.com/developers/integrations/check_references/` page

### Motivation

https://datadoghq.atlassian.net/browse/WEB-3284

### Preview

- Test https://docs-staging.datadoghq.com/david.jones/validate-categories/integrations/ loads the same integrations.
- Test local build still works despite installing lib that isn't always available

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by davidejones** [comment]
cc @hestonhoffman for visibility from our discussion

**Review by devindford — APPROVED** [review]

**Comment by hestonhoffman** [comment]
Should we add `layouts/shortcodes/integration_categories.md` to .gitignore?

File: layouts/shortcodes/integration_categories.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+- Supported OS::HP-UX
+- Supported OS::IBM i/OS
+- Supported OS::IBM z/OS
+- Supported OS::iOS
+- Supported OS::Linux
+- Supported OS::macOS
+- Supported OS::Solaris
+- Supported OS::Windows
+- Supported OS::Mac OS
+- Category::OS System
```
Comment (kayayarai) [inline layouts/shortcodes/integration_categories.md:70]: Is this different from line 34 `OS & System`?

**Comment by davidejones** [comment]
> Should we add `layouts/shortcodes/integration_categories.md` to .gitignore?

@hestonhoffman  So this was intentional but it really depends on how this shortcode is used

- if its used in committed files we need to make sure `layouts/shortcodes/integration_categories.md` exists so it doesn't throw non existant errors for users who won't have access to the assetlib
- If its only used in autogenerated files we can gitignore it as you said as it will generate with the other files

It would be better to do the latter if you can confirm that its only going to be used in that way?

File: layouts/shortcodes/integration_categories.md (hunk 1 above)
Comment (davidejones) [reply layouts/shortcodes/integration_categories.md:70]: I believe this is temporary while they are being corrected in upstream repos but yes this is a good point we don't want this to appear as a valid choice for new integrations

**Comment by hestonhoffman** [comment]
@davidejones Okay, it will be used in committed files, so let's keep it.

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**